packages/synchro-charts/src/components/sc-dial/sc-dial-base/sc-dial-svg.tsx

```diff
@@ -0,0 +1,201 @@
+import { Component, h, Prop, State, Watch, Element } from '@stencil/core';
+import { NO_VALUE_PRESENT } from '../../../constants';
+import { DataPoint, DataStream } from '../../../utils/dataTypes';
+import { round } from '../../../utils/number';
+import { getIcons } from '../../charts/common/annotations/iconUtils';
+import { Threshold } from '../../charts/common/types';
+import { DialSizeConfig } from '../utils/type';
+import { ColorConfigurations } from '../utils/util';
+
+const FONT_SIZE = 48;
+const ICON_SIZE = 24;
+const LABEL_SIZE = 24;
+const UNIT_SIZE = 24;
+const DIAL_THICKNESS = 34;
+const R = 138;
+const STROKE_WIDTH = 2;
+@Component({
+  tag: 'sc-dial-svg',
+  shadow: false,
+})
+export class ScDialSvg {
+  @Element() el: SVGGraphicsElement;
+  @Prop() percent: number;
+  @Prop() point?: DataPoint;
+  @Prop() breachedThreshold: Threshold;
+  @Prop() stream?: DataStream | null;
+  @Prop() size?: DialSizeConfig;
+  @Prop() significantDigits?: number;
+  @Prop() unit: string;
+
+  @State() sizeConfig: DialSizeConfig;
+
+  @State() labelBottomMiddleX: number = 0;
+
+  @State() iconLeftTopX: number = 0;
+  @State() iconLeftTopY: number = 0;
+
+  @State() showLabel: boolean = false;
+
+  componentWillLoad() {
+    this.onSizeChanged(this.size);
+  }
+
+  componentDidLoad() {
+    const DEFAULT = 0;
+    const label = this.el.querySelectorAll('text')[1];
+    const labelWidth = (label?.getBBox && label?.getBBox().width) || DEFAULT;
+    const labelHeight = (label?.getBBox && label?.getBBox().height) || DEFAULT;
+    const labelY = (label?.getBBox && label?.getBBox().y) || DEFAULT;
+
+    this.labelBottomMiddleX = R + (this.sizeConfig.iconSize - labelWidth) / 2;
+    this.iconLeftTopX = R - (this.sizeConfig.iconSize + labelWidth) / 2;
```

Comment:
and what happens if the label changes?

Reply:
The label and icon are displayed in the center of dial.